libs/react-spa/bff/src/lib/BffProvider.tsx

```diff
@@ -150,13 +196,13 @@ export const BffProvider = ({
       })
 
       window.location.href = bffUrlGenerator('/login', {
-        target_link_uri: window.location.href,
+        target_link_uri: loginQueryParams['target_link_uri'],
         ...(nationalId
           ? { login_hint: nationalId }
           : { prompt: 'select_account' }),
       })
     },
-    [bffUrlGenerator],
+    [bffUrlGenerator, loginQueryParams],
```

Comment:
This may not work as expected. Since `loginQueryParams` is memoized. I might have browsed around before calling switchUser, and end up on some old page.

Though I'm curious if this matches previous logic as I feel like at some point we always sent you to the homepage since you might not have access to the current page. Personally, I prefer this UX where you end up on the same page, even though sometimes you just see access denied.

Reply:
Ahh good point. I will update